### PR TITLE
add eye var

### PR DESCRIPTION
**Pull request by yajiedesign**
add batch_normalization

### CODE REVIEW

File: keras/backend/mxnet_backend.py [hunk 1]
```diff
@@ -1332,9 +1346,26 @@ def normalize_batch_in_training(x, gamma, beta,
...
     """
-    raise NotImplementedError
+    reduction_axes = _normalize_axis(reduction_axes, ndim(x))
+    if isinstance(x, KerasSymbol):
+        x = x.symbol
+
+    mean = mx.sym.mean(data=x, axis=reduction_axes, keepdims=True)
+
+    centered_input = mx.sym.broadcast_minus(lhs=x, rhs=mean)
+    var = mx.sym.mean(data=(centered_input ** 2), axis=reduction_axes, keepdims=True)
```
Comment (piiswrong) [inline keras/backend/mxnet_backend.py]: This doesn't seem right.

File: keras/backend/mxnet_backend.py [hunk 2]
```diff
@@ -1332,9 +1346,26 @@ def normalize_batch_in_training(x, gamma, beta,
     # Returns
         A tuple length of 3, `(normalized_tensor, mean, variance)`.
     """
-    raise NotImplementedError
+    reduction_axes = _normalize_axis(reduction_axes, ndim(x))
```
Comment (howard0su) [inline keras/backend/mxnet_backend.py]: axes is a typo

File: keras/backend/mxnet_backend.py [hunk 3]
```diff
@@ -1923,6 +1959,8 @@ def sparse_categorical_crossentropy(output, target, from_logits=False):
     and a target tensor, where the target is an integer tensor.
     """
     raise NotImplementedError
```
Comment (howard0su) [inline keras/backend/mxnet_backend.py]: why raise is still here?

File: keras/backend/mxnet_backend.py [hunk 4]
```diff
@@ -1339,17 +1346,10 @@ def normalize_batch_in_training(x, gamma, beta,
     # Returns
         A tuple length of 3, `(normalized_tensor, mean, variance)`.
     """
-    reduction_axes = _normalize_axis(reduction_axes, ndim(x))
     if isinstance(x, KerasSymbol):
```
Comment (howard0su) [inline keras/backend/mxnet_backend.py]: this is not needed. we don't support x is not a kerasSymbol.

File: keras/backend/mxnet_backend.py [hunk 5]
```diff
@@ -1339,17 +1346,10 @@ def normalize_batch_in_training(x, gamma, beta,
...
         x = x.symbol
-
-    mean = mx.sym.mean(data=x, axis=reduction_axes, keepdims=True)
-
-    centered_input = x - mean
-    var = mx.sym.mean(data=(centered_input ** 2), axis=reduction_axes, keepdims=True)
-    normal = mx.sym.BatchNorm(data=x)
-
-    return KerasSymbol(normal), KerasSymbol(normal), KerasSymbol(var)
+    normal, mean, var, = mx.sym.BatchNorm(data=x, gamma=gamma.symbol, beta=beta.symbol, full_output=True)
```
Comment (howard0su) [inline keras/backend/mxnet_backend.py]: no , after var?

File: keras/backend/mxnet_backend.py [hunk 6]
```diff
@@ -1332,9 +1346,19 @@ def normalize_batch_in_training(x, gamma, beta,
...
+    normal, mean, var = mx.sym.BatchNorm(data=x, gamma=gamma.symbol, beta=beta.symbol, output_mean_var=True)
+    return KerasSymbol(normal), KerasSymbol(mean), KerasSymbol(var)
 
 
+def batch_normalization(x, mean, var, beta, gamma, epsilon=1e-3):
+    """Apply batch normalization on x given mean, var, beta and gamma.
+    """
+    if isinstance(x, KerasSymbol):
+        x = x.symbol
+    return KerasSymbol(mx.sym.BatchNorm(x))
```
Comment (piiswrong) [inline keras/backend/mxnet_backend.py]: should be mx.sym.BatchNorm(data=x, gamma=gamma, beta=beta, moving_mean=mean, moving_var=var)
